src/encoded/static/components/gene.js

```diff
@@ -10,96 +10,144 @@ import { RelatedItems } from './item';
 import { ExperimentTable } from './typeutils';
 
 
-const Gene = (props) => {
-    const context = props.context;
-    const itemClass = globals.itemClass(context, 'view-detail key-value');
-
-    // Set up breadcrumbs
-    const crumbs = [
-        { id: 'Genes' },
-        {
-            id: <i>{context.organism.scientific_name}</i>,
-            query: `organism.scientific_name=${context.organism.scientific_name}`,
-            tip: `${context.organism.scientific_name}`,
-        },
-    ];
-
-    const crumbsReleased = (context.status === 'released');
-
-    return (
-        <div className={globals.itemClass(context, 'view-item')}>
-            <header>
-                <Breadcrumbs root="/search/?type=gene" crumbs={crumbs} crumbsReleased={crumbsReleased} />
-                <h2>{context.symbol} (<em>{context.organism.scientific_name}</em>)</h2>
-                <ItemAccessories item={context} />
-            </header>
-
-            <div className="panel">
-                <dl className={itemClass}>
-                    {/* TODO link to NCBI Entrez page? */}
-                    <div data-test="gendid">
-                        <dt>Entrez GeneID</dt>
-                        <dd>
-                            <a href={dbxrefHref('GeneID', context.geneid)}>{context.geneid}</a>
-                        </dd>
-                    </div>
-
-                    {/* TODO link to NADB page? */}
-                    <div data-test="symbol">
-                        <dt>Gene symbol</dt>
-                        <dd>{context.symbol}</dd>
-                    </div>
+class Gene extends React.Component {
+    constructor() {
+        super();
+
+        this.state = {
+            goIDs: [],
+        };
+        this.baseGOUrl = 'http://api.geneontology.org/api/bioentity/gene/{0}/function';
+    }
+
+    componentDidMount() {
+        const nadbIDs = [];
+        const uniprotIDs = [];
+        this.props.context.dbxrefs.forEach((dbxref) => {
+            if (dbxref.startsWith('WormBase:WBGene')) {
+                nadbIDs.push(dbxref.replace('WormBase:', 'WB:'));
+            } else if (dbxref.startsWith('FlyBase:FBgn')) {
+                nadbIDs.push(dbxref.replace('FlyBase:', 'FB:'));
+            } else if (dbxref.startsWith('MGI:')) {
+                nadbIDs.push('MGI:'.concat(dbxref));
+            } else if (dbxref.startsWith('UniProtKB:')) {
+                uniprotIDs.push(dbxref);
+            }
+        });
+        Promise.all((nadbIDs.length > 0 ? nadbIDs : uniprotIDs).map(
+            goID => this.context.fetch(
+                this.baseGOUrl.replace(/\{0\}/g, goID), {
+                    method: 'GET',
+                }
+            ).then((response) => {
+                if (response.ok) {
+                    return response.json();
+                }
+                return { numFound: 0 };
+            }).then((annotation) => {
+                if (annotation.numFound) {
```

Comment:
Just one little thing: numbers should always be compared with a number (our ESLint configuration). ESLint doesn’t seem to detect it, but that’s rule we have. So `numFound > 0`.

Reply:
Thank you. You've mentioned this before; but I forgot. Sorry. Fixed.